src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -1981,8 +1981,26 @@ public Flux<EquipmentModificationInfos> createShuntCompensator(UUID networkUuid,
             }));
     }
 
-    public Mono<Void> moveModifications(UUID groupUuid, UUID before, List<UUID> modificationsToMove) {
-        return Mono.fromRunnable(() -> networkModificationRepository.moveModifications(groupUuid, modificationsToMove, before));
+    public Flux<UUID> moveModifications(UUID groupUuid, UUID before, List<UUID> modificationsToMove) {
+        return Flux.fromStream(() -> networkModificationRepository.moveModifications(groupUuid, modificationsToMove, before).stream());
+    }
+
+    public Flux<UUID> duplicateModifications(UUID targetGroupUuid, List<UUID> modificationsToDuplicate) {
+        // This function cannot be @Transactional because we clone all modifications resetting their id to null,
+        // which is not allowed by JPA if we still stay in the same Tx.
+        List<ModificationEntity> newModificationList = new ArrayList<>();
+        List<UUID> modificationsInFailure = new ArrayList<>();
+        for (UUID modifyId : modificationsToDuplicate) {
+            Optional<ModificationEntity> clone = this.modificationRepository.findById(modifyId);
+            if (clone.isEmpty()) {
+                modificationsInFailure.add(modifyId);  // data no more available
+            } else {
+                clone.get().setId(null);
+                newModificationList.add(clone.get());
+            }
+        }
+        networkModificationRepository.saveModifications(targetGroupUuid, newModificationList);
+        return Flux.fromStream(modificationsInFailure::stream);
```

Comment:
Not consistent with the creation of a deferred stream but is no longer up to date